presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/PrestoInformer.java

```diff
@@ -23,32 +23,37 @@
 
 import java.io.IOException;
 import java.nio.file.Path;
-import java.nio.file.Paths;
 
 import static com.teradata.prestomanager.agent.AgentFileUtils.getFileProperty;
 import static java.lang.String.format;
 import static javax.ws.rs.core.MediaType.TEXT_PLAIN;
 import static javax.ws.rs.core.UriBuilder.fromUri;
 
-public class PrestoUtils
+public class PrestoInformer
 {
-    private static final Path PRESTO_CONFIG_FILE = Paths.get("/etc/presto/config.properties");
-    private static final Logger LOGGER = Logger.get(PrestoUtils.class);
-    private static final Client client = JerseyClientBuilder.createClient();
+    private static final Logger LOGGER = Logger.get(PrestoInformer.class);
 
-    private PrestoUtils() {}
+    private final Path configFile;
+    private final Client client;
 
-    public static int getPrestoPort()
+    @Inject
+    PrestoInformer(Client client, PrestoConfig config)
```

Comment:
OK, you were just mentioning in a different place that you wanted to avoid passing in so many arguments